.github/workflows/test.yml

```diff
@@ -5,18 +5,24 @@ name: Test
 
 on:
   pull_request:
+    paths-ignore:
+      - "**.md"
+      - "**.yml"
+      - "**.yaml"
+      - "!.github/workflows/test.yml"
   push:
+    paths-ignore:
+      - "**.md"
+      - "**.yml"
+      - "**.yaml"
+      - "!.github/workflows/test.yml"
+    branches-ignore:
+      - "dependabot/**"
+      - "pre-commit-ci-update-config"
+    tags: ["**"]
   workflow_dispatch:
 
 jobs:
-  pre-commit:
-    name: Run pre-commit
-    runs-on: ubuntu-20.04
-    steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-python@v3
-      - uses: pre-commit/action@v2.0.3
-
```

Comment:
If merged, I'd enable pre-commit.ci as done across the jupyterhub org to run these tests, and more:
- create PRs to update the pre-commit-config.yaml file with modern versions of the hooks
- add commits to PRs with autoformatting changes

We have tried it out in many repos now with great success with regards to the UX for misc contributors and core developers. There is a jupyterhub team-compass issue about it for a centralized discussion: https://github.com/jupyterhub/team-compass/issues/379

Reply:
To turn pre-commit.ci on for this repo, a JupyterHub github org admin should visit: https://github.com/organizations/jupyterhub/settings/installations/17782968